File: myLeetCode/String/541. Reverse String II.cpp
// 541. Reverse String II
// https://leetcode.com/problems/reverse-string-ii/

#include <iostream>

class Solution {
public:
    std::string reverseStr(std::string s, int k) {
        int n = s.length();
        
        for (int i = 0; i < n; i += 2 * k) {
            int left = i;
            int right = std::min(i + k - 1, n - 1);
            
            while (left < right) {
                std::swap(s[left], s[right]);
                ++left;
                --right;
            }
        }
        
        return s;
    }
};

int main() {
    Solution sol;
    std::string s = "abcdefg";
    int k = 2;
    
    std::string result = sol.reverseStr(s, k);
    
    std::cout << result << std::endl;
    
    return 0;
}

File: C++/代码随想录/String/541. Reverse String II.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        s = list(s)
        for i in range(0, len(s), 2 * k):
            s[i:i + k] = reversed(s[i:i + k])
        return ''.join(s)

sol = Solution()

# 测试用例
print(sol.reverseStr("abcdefg", 2))  # 应输出: "bacdfeg"
print(sol.reverseStr("abcd", 2))     # 应输出: "bacd"